### PR TITLE
Codechange: Simplify drawing of timetables

### PR DESCRIPTION
**Pull request by glx22**
## Motivation / Problem
Each timetable departure/arrival time is drawn using 2 strings, the abbreviation and the time.

## Description
Introduce `{COLOUR}` control code and use it to merge abbreviation and time into a single string.

## Limitations
EINTS needs to be updated.

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, gs_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

File: src/strings.cpp [hunk 1]
```diff
@@ -1618,6 +1618,12 @@ static char *FormatString(char *buff, const char *str_arg, StringParameters *arg
 				break;
 			}
 
+			case SCC_COLOUR: { // {COLOUR}
+				StringControlCode scc = (StringControlCode)(SCC_BLUE + args->GetInt32(SCC_COLOUR));
+				if (IsInsideMM(scc, SCC_BLUE, SCC_COLOUR) && (buff + Utf8CharLen(scc) < last)) buff += Utf8Encode(buff, scc);
```
Comment (PeterN) [inline src/strings.cpp]: This relies on SCC_COLOUR being placed after SCC_BLACK. May be better as `SCC_BLACK + 1`? Though this does all rely on the colours being in specific order in the first place, so may not matter.

**Comment by glx22** [comment]
It's ready, but I keep it as draft because it depends on https://github.com/OpenTTD/eints/pull/142

**Review by 2TallTyler — APPROVED** [review]
I assume you've tested this thoroughly. Code looks good to me. 

I wonder, idly of course since it's out of scope here, if there are other places that could use this simplified drawing method.

File: src/strings.cpp [hunk 2]
```diff
@@ -1627,6 +1627,12 @@ static void FormatString(StringBuilder &builder, const char *str_arg, StringPara
 				break;
 			}
 
+			case SCC_COLOUR: { // {COLOUR}
+				StringControlCode scc = (StringControlCode)(SCC_BLUE + args.GetInt32(SCC_COLOUR));
```
Comment (nielsmh) [inline src/strings.cpp:1631]: I'm getting a compile error on this line after the merge.
```
C:\Users\nielsm\Dev\OpenTTD\src\strings.cpp(1628): error C2660: 'StringParameters::GetInt32': function does not take 1 arguments
  C:\Users\nielsm\Dev\OpenTTD\src\strings_internal.h(91): note: see declaration of 'StringParameters::GetInt32'
  C:\Users\nielsm\Dev\OpenTTD\src\strings.cpp(1628): note: while trying to match the argument list '(StringControlCode)'
```

As far as I can tell, `StringParameters::GetInt32()` has never existed in a version that takes a parameter.

**Pull request merged**